### PR TITLE
Update options for test_installer_ccm.py

### PR DESCRIPTION
**Pull request by mellenburg**
-Minuteman enabled is a deprecated configuration option
-Collect AWS env vars in the same manner as everything else
-Add ability to set cluster size/composition

### CODE REVIEW

**Comment by mellenburg** [comment]
@cmaloney, @spahl  this is ready for review

**Comment by spahl** [comment]
please split into 3 commits.

**Comment by mellenburg** [comment]
@spahl done

File: test_util/test_installer_ccm.py [hunk 1]
```diff
@@ -214,15 +211,10 @@ def integration_test(
         region: string indicating AWS region in which cluster is running
         agent_list: string of comma separated agent addresses
         registry_host: string for address where marathon can pull test app
-        test_minuteman: if set to True then test for minuteman service
         test_dns_search: if set to True, test for deployed mesos DNS app
         ci_flags: optional additional string to be passed to test
 
     """
-    marker_args = '-m "not minuteman"'
```
Comment (cmaloney) [inline test_util/test_installer_ccm.py]: should also remove the marker from integration_test.py

File: test_util/test_installer_ccm.py [hunk 2]
```diff
@@ -339,6 +343,13 @@ def check_environment():
     options.ci_flags = os.getenv('CI_FLAGS', '')
     options.aws_access_key_id = os.environ.get('AWS_ACCESS_KEY_ID', '')
     options.aws_secret_access_key = os.environ.get('AWS_SECRET_ACCESS_KEY', '')
+    options.num_masters = int(os.environ.get('NUM_MASTERS', '1'))
+    options.num_agents = int(os.environ.get('NUM_AGENTS', '2'))
+
+    if options.host_list:
+        total_hosts = len(options.host_list)
+        assert total_hosts == 1 + options.num_masters + options.num_agents
```
Comment (cmaloney) [inline test_util/test_installer_ccm.py]: comment string on this assertion?

File: test_util/test_installer_ccm.py [hunk 3]
```diff
@@ -350,7 +361,9 @@ def main():
     vpc = None  # Set if the test owns the VPC
 
     if options.host_list is None:
-        vpc = make_vpc(use_bare_os=options.test_install_prereqs)
+        vpc = make_vpc(
+            count=1+options.num_masters+options.num_agents,
```
Comment (cmaloney) [inline test_util/test_installer_ccm.py]: Note we'll have to revisit this when we add public agents

File: test_util/test_installer_ccm.py [hunk 4]
```diff
@@ -257,13 +257,16 @@ def prep_hosts(ssh_runner, registry):
...
-def make_vpc(use_bare_os=False):
+def make_vpc(count, use_bare_os=False):
     """uses CCM to provision a test VPC of minimal size (3).
 
     Args:
+        count: number of instances to spin up
         use_bare_os: if True, vanilla AMI is used. If False, custom AMI is used
             with much faster prereq satisfaction time
     """
+    assert count >= 4, ('Cluster must be at least 4 hosts: '
```
Comment (cmaloney) [inline test_util/test_installer_ccm.py]: nit: This logic is a bit spread all over the place. Might make sense to centralize the notion of "I need a block of hosts" and have that class own initializing from either environment variables or building a new CCM VPC.

File: test_util/test_installer_ccm.py [hunk 5]
```diff
@@ -381,8 +394,17 @@ def make_runner(host_list):
     print("VPC hosts: {}".format(host_list))
     # use first node as bootstrap node, second node as master, all others as agents
     registry_host = local_ip[host_list[0]]
-    master_list = [local_ip[_] for _ in host_list[1:2]]
-    agent_list = [local_ip[_] for _ in host_list[2:]]
+    master_list = []
+    agent_list = []
+    print('Test/registry host public/private IP: ' + host_list[0] + '/' + registry_host)
+    for i in range(1, (options.num_masters+1)):
```
Comment (cmaloney) [inline test_util/test_installer_ccm.py]: master_list = host_list[1:options.num_masters+1]
agent_list = host_list[options.num_masters+1:options.num_agents+1]

print("Master IPs:", master_list)
print("Agent IPs:", agent_list)

Comment (mellenburg) [inline test_util/test_installer_ccm.py]: The goal was to print out the hosts in such a way that one could easily parse the local and public IP for a single node

Comment (cmaloney) [inline test_util/test_installer_ccm.py]: I don't see how that new formatting is any more difficult than the old one to parse, it's still arbitrary / hand written. If we want machine parsable, then should do something like `print(json.dumps({'master_list': master_list, 'agent_list': agent_list})`

File: test_util/test_installer_ccm.py (hunk 1 above)
Comment (mellenburg) [inline test_util/test_installer_ccm.py]: I dont see any usage of mark in integration_test.py currently?

File: test_util/test_installer_ccm.py (hunk 4 above)
Comment (mellenburg) [inline test_util/test_installer_ccm.py]: In that case, the code will no longer be cleanup, so I'll take it out of here for now and put it in another PR so we can get the other two basic commits in

File: test_util/test_installer_ccm.py (hunk 1 above)
Comment (cmaloney) [inline test_util/test_installer_ccm.py]: Apparently when minuteman was made non-conditional, that was removed but setting the marker wasn't. So yea, nothing to do there, my bad

**Comment by mellenburg** [comment]
@cmaloney this has been stripped down so it only touches the minuteman and aws values cleanup. Can it ship?

**Comment by cmaloney** [comment]
:ship:. I'll add to the next train.

**Pull request merged**